docs/src/index.md

```diff
@@ -29,7 +29,52 @@ F(t,y) = y
 The vectors `t` and `y` store the time and solution values at the
 corresponding times.
 
-You might find the basic interface limiting.  First of all, it stores
+## Solve interface
+
+`ODE.ode` only supports explicit differential equations defined as
+`y'=F(t,y)`, for more advenced uses consider using `ODE.solve`, which
+was designed to work with a variety of other types of initial value
+problems and is optimized for better performance.  First we have to
+define an initial value problem, in our case this is an explicit
+differential equation `y'=y` with inital data `y0=[1.0]` given at the
+time `t0=0.0`.
+
+```@example solve
+using ODE
+t0  = 0.0
+y0  = [1.0]
+F!(t,y,dy) = dy[1]=y[1]
+ode = ODE.ExplicitODE(t0,y0,F!)
+```
+
+Note that unlike in `ODE.ode` we now have to supply an in place
+function `F!` instead of an explicit function `F`.  We can solve the
+ODE problem `ode` by simply calling
+
+```@example solve
+sol = ODE.solve(ode, tstop = 1)
+```
```

Comment:
One more solution would be to have `maxiters` hard coded into the iterator and set to some reasonable finite number by default.  This would prevent any runaway integrations.